test/new-e2e/tests/containers/kindvm_test.go

```diff
@@ -8,24 +8,51 @@ package containers
 import (
 	"context"
 	"encoding/json"
+	"os"
 	"testing"
 
 	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/components"
+
+	"fmt"
 	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/runner"
 	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/infra"
 	"github.com/DataDog/test-infra-definitions/scenarios/aws/kindvm"
-
 	"github.com/pulumi/pulumi/sdk/v3/go/auto"
 	"github.com/stretchr/testify/suite"
 	"k8s.io/client-go/tools/clientcmd"
+	"strings"
 )
 
 type kindSuite struct {
 	k8sSuite
+	k8sVersion string
+	stackName  string
 }
 
 func TestKindSuite(t *testing.T) {
-	suite.Run(t, &kindSuite{})
+	stackConfigFromEnv := os.Getenv("E2E_STACK_PARAMS")
+
+	if strings.Contains(stackConfigFromEnv, "ddinfra:kubernetesVersion") {
+		// If kubernetes version is already specified in cli, run suite only for specified version
+		suite.Run(t, &kindSuite{})
+	} else {
+		var KubeVersions []string
+
+		versions := os.Getenv("KUBERNETES_VERSIONS")
+		if versions != "" {
+			KubeVersions = strings.Split(versions, ",")
+		} else {
+			KubeVersions = []string{"1.27", "1.23", "1.20", "1.19"}
+		}
```

Comment:
Could we use `flag`s instead of environment variables ?

```go
var kubernetesVersions = flag.String("kubernetes-versions", "1.27,1.23,1.20,1.19", "Comma-separated list of Kubernetes version to test")
```
outside of `TestKindSuite` function. And then:

```suggestion
		kubeVersions = strings.Split(kubernetesVersions, ",")
```